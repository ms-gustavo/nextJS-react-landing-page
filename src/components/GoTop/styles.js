import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 2rem;
    z-index: 6;
    right: 2rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    opacity: 0.7;
    border-radius: 50%;
  `}
`;
