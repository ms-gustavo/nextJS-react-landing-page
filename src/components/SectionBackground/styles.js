import styled, { css } from 'styled-components';

const containerBackgroundActivate = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: flex;
    align-items: center;
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};

    ${background && containerBackgroundActivate(theme)};

    min-height: 100vh;
  `}
`;
