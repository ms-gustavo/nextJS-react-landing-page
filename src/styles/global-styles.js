import { createGlobalStyle, css } from 'styled-components';

export const GlobalStyles = createGlobalStyle`

  @import url('https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@0,900;1,900&family=Open+Sans:wght@700&family=Poppins:ital,wght@0,300;1,300&display=swap');

*{
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html {
  font-size: 62.5%;
}

body{
  font-size: 1.6rem;
  font-family: ${({ theme }) => theme.font.family.default}
}

h1, h2, h3, h4, h5, h6 {
  font-family: ${({ theme }) => theme.font.family.secondary}
}
`;
