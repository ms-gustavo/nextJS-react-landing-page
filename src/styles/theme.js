export const theme = {
  colors: {
    mainBg: 'purple',
    secondaryBg: 'yellow',
  },
  fonts: {},
  spacings: {},
};
