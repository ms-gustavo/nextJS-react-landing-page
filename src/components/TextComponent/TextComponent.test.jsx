import { screen } from '@testing-library/react';
import { TextComponent } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<TextComponent/>', () => {
  it('should render a text', () => {
    renderTheme(<TextComponent>Children</TextComponent>);
    expect(screen.getByText('Children')).toBeInTheDocument();
  });
  it('should match snapShot', () => {
    const { container } = renderTheme(<TextComponent>Children</TextComponent>);
    expect(container).toMatchSnapshot();
  });
});
