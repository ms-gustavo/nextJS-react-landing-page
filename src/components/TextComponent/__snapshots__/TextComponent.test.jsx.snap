// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent/> should match snapShot 1`] = `
.c0 {
  font-size: 2.4rem;
}

<div>
  <div
    class="c0"
  >
    Children
  </div>
</div>
`;
